src/rider/main/kotlin/me/fornever/avaloniarider/AvaloniaPreviewerWindow.kt

```diff
@@ -0,0 +1,38 @@
+package me.fornever.avaloniarider.me.fornever.avaloniarider
+
+import me.fornever.avaloniarider.FrameMessage
+import java.awt.Color
+import java.awt.FlowLayout
+import java.awt.image.BufferedImage
+import javax.swing.ImageIcon
+import javax.swing.JFrame
+import javax.swing.JLabel
+
+class AvaloniaPreviewerWindow : JFrame("Avalonia Previewer") {
+
+    companion object {
+        private fun fromByte(b: Byte): Int = b.toInt() and 0xFF
+    }
+
+    private val content = JLabel()
+    init {
+        layout = FlowLayout()
+        contentPane.add(content)
+    }
+
+    fun drawFrame(frame: FrameMessage) {
+        @Suppress("UndesirableClassUsage")
+        val img = BufferedImage(frame.width, frame.height, BufferedImage.TYPE_INT_RGB)
```

Comment:
Changed to single img usage. Made it static since there will be only one window, is it right?
Added function fillImage to clear it before loading new image.
https://github.com/ForNeVeR/AvaloniaRider/pull/7/commits/783a9d74ef1b78022f56b32add98b402d76b0237

Reply:
I don't think that static here is good, let's not make it static.

Overusing statics is one of common problems with code of novice programmers. Generally you should view the code from the architectural point of view, or, in this case, from a lifetime point of view.

Each `AvaloniaPreviewerWindow` should contain one and only one `BufferedImage`, right? That means that `BufferedImage` may be an instance field of `AvaloniaPreviewerWindow`, because **lifetimes** of these objects are identical.

If you don't like my philosophical reasoning, then here's another, real reason: in future, there will be multiple `AvaloniaPreviewerWindow`s (in case user opens multiple editor tabs, each with a separate designer window). So `BufferedImage` shouldn't be static.

Reply:
Yes, I do agree with your philosophy. Made it non-static in new commit https://github.com/ForNeVeR/AvaloniaRider/pull/7/commits/8297e0c7f768b3d1c328bd1a21992b4ee168d69f

Reply:
Thanks for a profound explanation!